wikiextractor/extract.py

```diff
@@ -194,9 +196,8 @@ def compact(text, mark_headers=False):
 
             headers[lev] = title
             # drop previous headers
-            for i in headers.keys():
-                if i > lev:
-                    del headers[i]
+            for _ in set(i for i in headers if i > lev):
```

Comment:
What is the purpose of re-writing the `for` loop like this? Why a set? Why waste the time of creating a new iterable in the first place? Also, underscores are typically used as the variable of iteration only when the variable is not used inside the `for` loop, which is not the case here. I would write this as:

```python
for key in headers:
    if key > lev:
        del headers[key]
```

Reply:
Deleting the key in the for loop raises an error:
`RuntimeError: dictionary changed size during iteration`